rust/connlib/clients/shared/src/messages.rs

```diff
@@ -32,7 +31,7 @@ pub struct ConnectionDetails {
 
 #[derive(Debug, Deserialize, Serialize, Clone)]
 pub struct Connect {
-    pub gateway_rtc_session_description: RTCSessionDescription,
+    pub gateway_rtc_session_description: RTCIceParameters,
```

Comment:
@AndrewDryga we should change the name of the `*rtc_session_description` fields in the portal.

I think the problem is that we exposed too much details here. Ee should call it something like `connection_parameters`.

Reply:
@conectado maybe `ice_state`? `parameters` terms feels a bit abstract and overloaded

Reply:
But it's not so much about `state` since it's such parameters for the connection.

We could use `ice_session`.

Reply:
My preference would be to stop using `webrtc` internal types for this altogether. We could simply inline the type and move the parameters to the `Connect` messages. That way we don't need to name the type and get full control over the serialization (which is important for backwards-compatibility).

Reply:
the internal fields are:
* `username_fragment`
* `password`
* `ice_lite`(<- this can be ignored and set to false for now)

I'm not sure we will keep those fields around.

But we can still have our own type for session information instead that way if we change the number and kind of parameters we don't lose compatibility with the portal. Yet, for the portal, the type itself is opaque, just a string so we can use whatever here and for now still use `webrtc` internal type.

Reply:
For example, do we need to transfer the state whether we use `ice_lite`? Isn't that something we can hardcode on either end?

We can probably trim this down to have a field `ice_credentials` with two further fields `ufrag` and `password`. If we wanted to, we could even deterministically create those on both ends, then we don't have to send them over but that is something we can always do at a later point.

Reply:
> But we can still have our own type for session information instead that way if we change the number and kind of parameters we don't lose compatibility with the portal.

Perhaps `payload` would be a good name then? For the portal it should be opaque and just forwarded to the gateway.

Reply:
I like `payload` cc @AndrewDryga 

> We can probably trim this down to have a field ice_credentials with two further fields ufrag and password. If we wanted to, we could even deterministically create those on both ends, then we don't have to send them over but that is something we can always do at a later point.

I don't think we would even need them in the future: since we trust the websocket and we can multiplex based on the `client_id`.

Reply:
I like `payload` or `metadata` too, whatever you guys think is more appropriate.